### PR TITLE
Add l1 status to citrea_syncStatus

### LINKED ISSUE #1186: Sync status should also show l1 sync status
Currently we have an rpc call `citrea_syncStatus` which shows l2 syncing status of full nodes to sequencer.

It would be also nice to see the status of l1 sync.

- [ ] In the rpc find a way to access which l1 block the node last synced
- [ ] Update the `CitreaStatus` in a way that includes l1 sync status
- [ ] Update rpc endpoint 
- [ ] Update tests using the rpc endpoint and test l1 sync feature

### PR DESCRIPTION
**Pull request by exeokan**
# Description

Update the citrea_syncStatus rpc to also show l1 sync status.

## Linked Issues
- Fixes #1186

### CODE REVIEW

File: crates/ethereum-rpc/src/lib.rs [hunk 1]
```diff
@@ -627,20 +635,41 @@ fn register_rpc_methods<C: sov_modules_api::Context, Da: DaService>(
...
-                    Ok::<CitreaStatus, ErrorObjectOwned>(CitreaStatus::Syncing(SyncStatus {
-                        synced_block_number,
-                        head_block_number,
-                    }))
+                let l1_synced_block_number = match ethereum.ledger_db.get_last_scanned_l1_height() {
+                    Ok(Some(slot_number)) => slot_number.0,
+                    Ok(None) => 0,
+                    Err(e) => return Err(to_jsonrpsee_error_object("PROVER_ERROR", e)),
+                };
+                let l1_head_block_number = match ethereum.da_service.get_head_block_header().await {
```
Comment (eyusufatik) [inline crates/ethereum-rpc/src/lib.rs:649]: i don't think this is updated in ledger db

File: crates/ethereum-rpc/src/lib.rs [hunk 2]
```diff
@@ -627,20 +635,41 @@ fn register_rpc_methods<C: sov_modules_api::Context, Da: DaService>(
...
                     Ok(None) => 0u64,
                     Err(e) => return Err(e),
                 };
 
-                if synced_block_number < head_block_number {
-                    Ok::<CitreaStatus, ErrorObjectOwned>(CitreaStatus::Syncing(SyncStatus {
-                        synced_block_number,
-                        head_block_number,
-                    }))
+                let l1_synced_block_number = match ethereum.ledger_db.get_last_scanned_l1_height() {
```
Comment (eyusufatik) [inline crates/ethereum-rpc/src/lib.rs:644]: for the sequencer this table is never updated. however we were already considering disabling healthcheck endpoint for sequencer. so do it

File: crates/ethereum-rpc/src/lib.rs (hunk 1 above)
Comment (eyusufatik) [inline crates/ethereum-rpc/src/lib.rs:649]: use last finalized height

File: crates/ethereum-rpc/src/lib.rs [hunk 3]
```diff
@@ -613,7 +621,7 @@ fn register_rpc_methods<C: sov_modules_api::Context, Da: DaService>(
                     .block_number()
                     .await;
```
Comment (eyusufatik) [inline crates/ethereum-rpc/src/lib.rs:623]: we have two requests in this function now.

1. to the sequencer
2. to the da layer.

we can do both async, and await on them, lowering time needed to execute this function

File: crates/ethereum-rpc/src/lib.rs [hunk 4]
```diff
@@ -33,18 +35,23 @@ pub struct SyncStatus {
...
 #[derive(Clone, Debug, serde::Serialize, serde::Deserialize, PartialEq)]
-#[serde(rename_all = "camelCase")]
-pub enum CitreaStatus {
+pub enum LayerStatus {
     Synced(u64),
     Syncing(SyncStatus),
 }
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct CitreaStatus {
```
Comment (eyusufatik) [inline crates/ethereum-rpc/src/lib.rs:45]: Can you rename this to SyncStatus?

**Review by eyusufatik — CHANGES_REQUESTED** [review]

File: crates/ethereum-rpc/src/lib.rs (hunk 3 above)
Comment (eyusufatik) [reply crates/ethereum-rpc/src/lib.rs:623]: @kpp @rakanalh in tokio, what's the equivalent of JavaScript's Promise.all ?

Comment (rakanalh) [reply crates/ethereum-rpc/src/lib.rs:623]: https://docs.rs/futures/0.3.5/futures/future/fn.join_all.html

Comment (rakanalh) [reply crates/ethereum-rpc/src/lib.rs:623]: or:
https://docs.rs/tokio/latest/tokio/macro.join.html

Comment (exeokan) [reply crates/ethereum-rpc/src/lib.rs]: done but I cloned ethereum.sequencer_client, if it is bad practice I can find a workaround

File: crates/ethereum-rpc/src/lib.rs (hunk 2 above)
Comment (eyusufatik) [reply crates/ethereum-rpc/src/lib.rs:652]: I got confused about the healthcheck. this is a different rpc method. 

still sending this to the sequencer will be problematic

Comment (eyusufatik) [reply crates/ethereum-rpc/src/lib.rs:652]: and this is already only registered if not sequencer. cool

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**